_resources/port1.0/group/meson-1.0.tcl

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8; mode: tcl; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 4; truncate-lines: t -*- vim:fenc=utf-8:et:sw=4:ts=4:sts=4
+#
+# Copyright (c) 2017 The MacPorts Project
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+# 3. Neither the name of Apple Computer, Inc. nor the names of its
+#    contributors may be used to endorse or promote products derived from
+#    this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+# This PortGroup supports the meson build system
+#
+# Usage:
+#
+# PortGroup meson 1.0
+#
+# Currently the only configuration option is
+#   meson.build_dir ${...}
+# which defaults to ${workpath}/build,
+# but there should be no need to change that.
+
+#---------
+# WARNING:
+#---------
+#
+# Meson's install_name currently seems to be broken, so workarounds might be needed to make ports actually work.
+# See: https://github.com/mesonbuild/meson/issues/2121
+
+
+# meson builds need to be done out-of-source
+options meson.build_dir 
+default meson.build_dir     {${workpath}/build}
+
+depends_build-append        port:meson \
+                            port:ninja
```

Comment:
`depends_skip_archcheck meson ninja` should probably be added since the architecture of the build tools probably doesn't matter, so long as it runs on the build system.

There are a couple ports, like textmate2 and xcbuild, that already use ninja without meson. Could we create a ninja portgroup for those ports to use, and could we then include the ninja portgroup in the meson portgroup to reduce code duplication?

Reply:
Good catch, thank you. There are plenty of places where perl5 is built universal, most likely by accident.

I don't have any strong opinion about ninja portgroup. If you find it useful, we can do it. In this case it hardly does anything else than adding a dependency and setting the binary, but why not.

Reply:
Well textmate2 also sets `-v` to build verbosely; that could happen in a portgroup. And whatever ends up being necessary to support universal builds could go there too.